chainer/function_node.py

```diff
@@ -245,6 +245,12 @@ def apply(self, inputs):
             outputs = self.forward(in_data)
             assert type(outputs) is tuple
 
+        # Convert NumPy scalars to numpy.ndarray
+        outputs = tuple([
```

Comment:
I think this convert is redundant.
Currently, a `forward` function must return `ndarray` ,

Reply:
How about moving to https://github.com/chainer/chainer/pull/3744/files#diff-3480da470ca3b18f6b879875999e6206L253

Reply:
It that for performance?
Well, indeed I confirmed this PR degrades the performance of `apply` by >5%.

I'll update so that it will generate a nice error with debug mode.

Reply:
Thanks